internal/store/driver.go

```diff
@@ -642,8 +844,424 @@ func (d *Driver) calculateHTTPSEdges() map[string]ingressv1alpha1.HTTPSEdge {
+	inboundActions := Actions{}
+	outboundActions := Actions{}
+	expressions := []string{}
+	pathPrefixMatches := []string{}
+
+	// NOTE: matches are only defined on requests, and fitlers are only triggered by matches,
+	// but some fitlers define transformations on responses, so we need to define matches on both
+	// Policy.Inbound and Policy.Outbound when possible to work with ngrok's system
+	for _, match := range rule.Matches {
+		if match.Path != nil {
+			if match.Path.Type != nil {
+				switch *match.Path.Type {
+				case gatewayv1.PathMatchExact:
+				case gatewayv1.PathMatchPathPrefix:
+					if match.Path.Value != nil {
+						pathPrefixMatches = append(pathPrefixMatches, *match.Path.Value)
+					}
+				case gatewayv1.PathMatchRegularExpression:
+					return nil, errors.NewErrorNotFound(fmt.Sprintf("unsupported match type PathMatchType %v found", *match.Path.Type))
+				default:
+					return nil, errors.NewErrorNotFound(fmt.Sprintf("Unknown match type PathMatchType %v found", *match.Path.Type))
+				}
+			}
+		}
+
+		if match.Method != nil {
+			d.log.Error(fmt.Errorf("Unsupported match type"), "Unsupported match type", "HTTPMethod", *match.Method)
+		}
+
+		if len(match.Headers) > 0 {
+			d.log.Error(fmt.Errorf("Unsupported match type"), "Unsupported match type", "HTTPHeaderMatch", match.Headers)
+		}
+
+		if len(match.QueryParams) > 0 {
+			d.log.Error(fmt.Errorf("Unsupported match type"), "Unsupported match type", "HTTPQueryParamMatch", match.QueryParams)
+		}
+	}
+
+	responseHeaders := make(map[string]string)
+	for _, filter := range rule.Filters {
+		switch filter.Type {
+		case gatewayv1.HTTPRouteFilterRequestRedirect:
+			// NOTE: request redirect is a special case, and is subject to change
+			err := d.handleRequestRedirectFilter(filter.RequestRedirect, pathPrefixMatches, &inboundActions, responseHeaders)
+			if err != nil {
+				return nil, err
+			}
+		case gatewayv1.HTTPRouteFilterRequestHeaderModifier:
+			err := d.handleHTTPHeaderFilter(filter.RequestHeaderModifier, &inboundActions, nil)
+			if err != nil {
+				return nil, err
+			}
+		case gatewayv1.HTTPRouteFilterResponseHeaderModifier:
+			err := d.handleHTTPHeaderFilter(filter.ResponseHeaderModifier, &outboundActions, responseHeaders)
+			if err != nil {
+				return nil, err
+			}
+		case gatewayv1.HTTPRouteFilterURLRewrite:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		case gatewayv1.HTTPRouteFilterRequestMirror:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		case gatewayv1.HTTPRouteFilterExtensionRef:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		default:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unknown filter HTTPRouteFilterType %v found", filter.Type))
```

Comment:
`errors.NewErrorNotFound` doesn't seem quite appropriate here, wdyt?

Reply:
I'll cut a new ticket to add new errors